photutils/psf/models.py

```diff
@@ -4,17 +4,428 @@
+        model.
+
+    ikwargs : dict, optional
+
+        Additional optional keyword arguments to be passed directly to the
+        `compute_interpolator` method. See `compute_interpolator` for more
+        details.
+
+    """
+    flux = Parameter(description='Intensity scaling factor for image data.',
+                     default=None)
+    x_0 = Parameter(description='X-position of a feature in the image in '
+                    'the output coordinate grid on which the model is '
+                    'evaluated.', default=0.0)
+    y_0 = Parameter(description='Y-position of a feature in the image in '
+                    'the output coordinate grid on which the model is '
+                    'evaluated.', default=0.0)
+
+    def __init__(self, data, flux=flux.default,
+                 x_0=x_0.default, y_0=y_0.default,
+                 normalize=False, correction_factor=1.0,
+                 origin=None, fill_value=0.0, ikwargs={}):
+        self._fill_value = fill_value
+        self._img_norm = None
+        self._normalization_status = 0 if normalize else 2
+        self._store_interpolator_kwargs(ikwargs)
+
+        if correction_factor <= 0:
+            raise ValueError("'correction_factor' must be strictly positive.")
+        self._correction_factor = correction_factor
+
+        self._data = np.array(data, copy=True, dtype=np.float64)
+
+        if not np.all(np.isfinite(self._data)):
+            raise ValueError("All elements of input 'data' must be finite.")
+
+        # set input image related parameters:
+        self._ny, self._nx = self._data.shape
+        self._shape = self._data.shape
+        if self._data.size < 1:
+            raise ValueError("Image data array cannot be zero-sized.")
+
+        # set the origin of the coordinate system in image's pixel grid:
+        self.origin = origin
+
+        if flux is None:
+            if self._img_norm is None:
+                self._img_norm = self._compute_raw_image_norm(self._data)
+            flux = self._img_norm
+
+        self._compute_normalization(normalize)
+
+        super(FittableImageModel2D, self).__init__(flux, x_0, y_0)
+
+        # initialize interpolator:
+        self.compute_interpolator(ikwargs)
+
+    def _compute_raw_image_norm(self, data):
+        """
+        Helper function that computes the uncorrected inverse normalization
+        factor of input image data. This quantity is computed as the
+        *absolute value* of the *sum of all pixel values*.
+
+        .. note::
+            This function is intended to be overriden in a subclass if one
+            desires to change the way the normalization factor is computed.
+
+        """
+        return np.abs(np.sum(self._data, dtype=np.float64))
+
+    def _compute_normalization(self, normalize):
+        """
+        Helper function that computes the inverse normalization factor of the
+        original image data. This quantity is computed as the *absolute value*
```

Comment:
Can you clarify why this is the absolute value instead of the straight-sum, @mcara?  That seems very counter-intuitive to me.  In the case of noisy data, or sampling far into the wings of a PSF, you *want* the negative noise spikes to cancel out the positive ones.  I also think no one would expect this as it's not standard for similar codes that I know of, so it would trip a lot of people up.